src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java

```diff
@@ -205,6 +240,90 @@ protected KualiDecimal calculateTotalAmountForLines(List<ConcurStandardAccountin
                 .reduce(KualiDecimal.ZERO, KualiDecimal::add);
     }
 
+    protected void addOriginEntriesForCorpCardPersonalExpenseCreditLines(Consumer<OriginEntryFull> entryConsumer) {
+        if (consolidatedCorpCardPersonalExpenseCredits.isEmpty()) {
+            return;
+        }
+        
+        for (List<ConcurStandardAccountingExtractDetailLine> corpCardPersonalCredits : consolidatedCorpCardPersonalExpenseCredits.values()) {
+            KualiDecimal totalSubGroupAmount = calculateTotalAmountForLines(corpCardPersonalCredits);
+            if (totalSubGroupAmount.isNonZero()) {
+                ConcurStandardAccountingExtractDetailLine firstCreditLine = corpCardPersonalCredits.get(0);
+                addOriginEntryForCorpCardPersonalExpense(
+                        entryConsumer, firstCreditLine, collectorHelper.getPrepaidOffsetAccountNumber(), StringUtils.EMPTY,
+                        collectorHelper.getPrepaidOffsetObjectCode(), totalSubGroupAmount);
+            }
+        }
+    }
+
+    protected void addOriginEntriesForCorpCardPersonalExpenseDebitLines(Consumer<OriginEntryFull> entryConsumer) {

```

Comment:
This is a pretty long function,  can it be refactored into some smaller functions?